### PR TITLE
Update pnpm workspace example in docs to avoid edge cases

### PR DESCRIPTION
**Pull request by sbussard**
The previous approach does not seem stable .

It was working with initial tests but now it's not, and I'd hate for someone to encounter this sort of snag based on my contribution.

Here's the type of error I was seeing

```
RR_PNPM_WORKSPACE_PKG_NOT_FOUND  In ../../../../../../../../external/aspect_rules_js+/js/private/test/image: "@mycorp/pkg-a@workspace:*" is in the dependencies but no package named "@mycorp/pkg-a" is present in the workspace
```

This happens if you add anything like - `'./**/*'` , `- '**'` , or `- '*'`  to workspace packages. It doesn't seem to matter if you have `- !..` , it still tries to look there.

So this new approach is to manually add project dirs and their subfolders. Easy enough.

```
- 'experimental/**'
- 'users/**'
- 'some_deeply_nested_project/**'
```

### CODE REVIEW

**Comment by sbussard** [comment]
This error could be due to the fact that I had removed `verify_node_modules_ignored = "//:.bazelignore"`  

https://bazelbuild.slack.com/archives/CEZUUKQ6P/p1739851043050399

But this PR is here just in case the problem is something else. 
I'll let the team decide what's best

File: docs/README.md [hunk 1]
```diff
@@ -109,21 +109,15 @@ $ bazel fetch @npm//...
...
-  - '*'
-  # Include all subdirectories at any depth
-  - '**/*'
-  # Exclude node_modules folders anywhere in the tree
-  - '!**/node_modules/**'
-  # Exclude node_modules folder in root
-  - '!node_modules'
-  # Exclude anything inside bazel dirs
-  - '!bazel-*/**'
+  - 'deeply_nested_project/**'
```
Comment (jbedard) [inline docs/README.md:118]: WDYT of `apps/**`, `packages/**` (doublestars), `tools/*` (single star) and `.`?

Comment (sbussard) [reply docs/README.md:118]: I personally don't prefer to organize top level projects by language or platform, but I know that's more familiar to the community, which will help with adoption. The one caveat is that I don't believe the `.` is necessary.

File: docs/README.md [hunk 2]
```diff
@@ -109,21 +109,15 @@ $ bazel fetch @npm//...
 
 ### Using pnpm workspaces
 
-Here's an example `pnpm-workspace.yaml` file which will automatically discover all packages in the repository, based on the existence of a `package.json` file.
+Here's an example `pnpm-workspace.yaml` file which will find all projects listed under `deeply_nested_project`, `users`, and `experimental`, based on the existence of a `package.json` file.
```
Comment (jbedard) [inline docs/README.md:112]: This line is out of date now

File: docs/README.md [hunk 3]
```diff
@@ -109,21 +109,15 @@ $ bazel fetch @npm//...
...
+
 Make sure to place this file at the root of the repository.
```
Comment (sbussard) [inline docs/README.md:114]:
```suggestion
Here's an example `pnpm-workspace.yaml` file which will find all projects under `apps` or `packages` at any depth, and anything directly under `tools`, based on the existence of a `package.json` file.

Make sure to place this file at the root of the repository.
```

File: docs/README.md (hunk 2 above)
Comment (sbussard) [reply docs/README.md]: My apologies. It's now updated.

**Review by jbedard — APPROVED** [review]
